src/measure_theory/conditional_expectation.lean

```diff
@@ -0,0 +1,242 @@
+/-
+Copyright (c) 2021 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne
+-/
+
+import measure_theory.lp_space
+
+/-! # Conditional expectation
+
+The conditional expectation will be defined for functions in `L²` by an orthogonal projection into
+a complete subspace of `L²`. It will then be extended to `L¹`.
+
+For now, this file contains only the definition of the subspace of `Lᵖ` containing functions which
+are measurable with respect to a sub-σ-algebra, as well as a proof that it is complete.
+
+-/
+
+noncomputable theory
+open topological_space measure_theory.Lp filter
+open_locale nnreal ennreal topological_space big_operators measure_theory
+
+namespace measure_theory
+
+/-- A function `f` verifies `ae_measurable' m f μ` if it is `μ`-a.e. equal to an `m`-measurable
+function. This is similar to `ae_measurable`, but the `measurable_space` structures used for the
+measurability statement and for the measure are different. -/
+def ae_measurable' {α β} [measurable_space β] (m : measurable_space α) {m0 : measurable_space α}
+  (f : α → β) (μ : measure α) :
+  Prop :=
+∃ g : α → β, @measurable α β m _ g ∧ f =ᵐ[μ] g
+
+namespace ae_measurable'
+
+variables {α β 𝕜 : Type*} {m m0 : measurable_space α} {μ : measure α}
+  [measurable_space β] [measurable_space 𝕜] {f g : α → β}
+
+lemma congr (hf : ae_measurable' m f μ) (hfg : f =ᵐ[μ] g) : ae_measurable' m g μ :=
+by { obtain ⟨f', hf'_meas, hff'⟩ := hf, exact ⟨f', hf'_meas, hfg.symm.trans hff'⟩, }
+
+lemma add [has_add β] [has_measurable_add₂ β] (hf : ae_measurable' m f μ)
+  (hg : ae_measurable' m g μ) :
+  ae_measurable' m (f+g) μ :=
+begin
+  rcases hf with ⟨f', h_f'_meas, hff'⟩,
+  rcases hg with ⟨g', h_g'_meas, hgg'⟩,
+  exact ⟨f' + g', @measurable.add α m _ _ _ _ f' g' h_f'_meas h_g'_meas, hff'.add hgg'⟩,
+end
+
+lemma const_smul [has_scalar 𝕜 β] [has_measurable_smul 𝕜 β] (c : 𝕜) (hf : ae_measurable' m f μ) :
+  ae_measurable' m (c • f) μ :=
+begin
+  rcases hf with ⟨f', h_f'_meas, hff'⟩,
+  refine ⟨c • f', @measurable.const_smul α m _ _ _ _ _ _ f' h_f'_meas c, _⟩,
+  exact eventually_eq.fun_comp hff' (λ x, c • x),
+end
+
+end ae_measurable'
+
+variables {α β γ E E' F F' G G' H 𝕜 𝕂 : Type*} {p : ℝ≥0∞}
+  [is_R_or_C 𝕜] -- 𝕜 for ℝ or ℂ
```

Comment:
Do you really use the version without the measurable space structure? If not, you could just use `𝕜` and drop `𝕂`.